security/webauthn/src/test/java/AbstractWebAuthnTest.java

```diff
@@ -0,0 +1,177 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.restassured.RestAssured;
+import io.restassured.filter.Filter;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpMethod;
+
+public abstract class AbstractWebAuthnTest {
+
+  protected abstract RestService getApp();
+
+  private static Vertx vertx;
+
+  private static HttpClient httpClient;
+
+  private String registerUrl = "/q/webauthn/register";
+  private String loginUrl = "/q/webauthn/login";
+  private String logOutUrl = "/q/webauthn/logout";
+
+  private String publicAPIUrl = "/api/public";
+  private String publicMeAPIUrl = "/api/public/me";
+  private String userAPIUrl = "/api/users/me";
+  private String adminAPIUrl = "/api/admin";
+
+  private String userName = "Roosvelt";
+  private static Filter cookieFilter;
+
+  @BeforeAll
+  public static void setup() {
+    vertx = Vertx.vertx();
+    httpClient = vertx.createHttpClient();
+    cookieFilter = new CookieFilter();
+    RestAssured.given().then().statusCode(200);
+
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    if (httpClient != null && vertx != null) {
+      httpClient.close();
+      vertx.close();
+    }
+  }
+
+  @Test
+  public void checkLogoutInitial() {
+    verifyLoggedOut(cookieFilter);
+  }
+
+  @Test
+  public void checkAdminAPIWithoutUser() {
+    RestAssured.given()
+      .redirects().follow(false)
+      .get(adminAPIUrl)
+      .then()
+      .statusCode(302);
+  }
+
+  @Test
+  public void checkMeAPIWithoutUser() {
+    RestAssured.given()
+      .redirects().follow(false)
+      .get(userAPIUrl)
+      .then()
+      .statusCode(302);
+  }
+
+  @Test
+  public void checkPublicAPI() {
+    RestAssured.given()
+      .get(publicAPIUrl)
+      .then()
+      .statusCode(200)
+      .body(Matchers.is("public"));
+  }
+
+  @Test
+  public void testRegisterWebAuthnUser() {
+
+    httpClient.request(HttpMethod.POST, registerUrl)
+      .compose(httpClientRequest -> {
+        httpClientRequest.putHeader("Content-Type", "text/plain")
+          .write(userName);
+        return httpClientRequest.send();
+      }).onSuccess(httpClientResponse -> {
```

Comment:
yes, you were right and this is now a problem because verifyLoggedIn wasn't called and when I've fixed it it seems verifyLoggedIn fails : 
`esponse body doesn't match expectation.
Expected: is "Roosvelt"
  Actual: <not logged in>`

Reply:
I've already solved that with the introduction of the new methods and MyWebAuthnHardware class